### PR TITLE
fix(update-modal): render release videos as an inline player

### PR DESCRIPTION
**Pull request by chhoumann**
## Summary

Release notes embed videos the GitHub-native way: a bare `github.com/user-attachments/...` URL on its own line (renders as an inline player on GitHub) plus a clickable play-button thumbnail as fallback. Obsidian's markdown renderer showed that bare URL as an ugly raw link in the update modal.

Obsidian's renderer can play these URLs natively, so the modal now renders an **actual inline video player**:

- a bare user-attachments URL line becomes `<video controls preload="metadata">` with the URL as `src`;
- a markdown thumbnail (`[alt](video-url)`) linking to the same video is folded into the player's **poster** instead of rendering as a duplicate visual;
- lone thumbnails (no matching bare URL) and non-attachment URLs are untouched.

## Verification

- 5 unit tests for the transform (`src/gui/UpdateModal/UpdateModal.test.ts`)
- Verified live in the dev vault against the real 2.13.0 release body: the modal shows a working player (poster + native controls, metadata loaded, 1920×1080 / 0:56) where the raw URL used to be — confirmed `user-attachments` URLs stream fine in Obsidian's `<video>` (302 → signed CDN URL, range requests OK)
- `bun run build` (tsc) and `bun lint` clean

## Release impact

Cosmetic/UX fix, rides the next release; no migration or settings impact.

## Summary by CodeRabbit

* **New Features**
  * Release notes now render bare video URLs as embedded playable video players; matching thumbnail links are used as poster images. Adjacent "poster" comments can override the poster. Non-matching links and unrelated content remain unchanged; repeated URLs render multiple players.

* **Tests**
  * Added comprehensive tests covering rendering, poster selection, comment handling, and edge cases.

### CODE REVIEW

File: src/gui/UpdateModal/UpdateModal.ts [hunk 1]
```diff
@@ -51,6 +51,38 @@ export async function getReleaseNotesAfter(
...
+		const trimmed = lines[i].trim();
+		if (!USER_ATTACHMENT_VIDEO_URL.test(trimmed)) {
+			result.push(lines[i]);
+			continue;
+		}
+
+		const linkedElsewhere = lines.some(
+			(other, j) => j !== i && other.includes(trimmed)
+		);
+		if (linkedElsewhere) continue;
```
[the remark this answers is not in the document]
Comment (chhoumann) [reply src/gui/UpdateModal/UpdateModal.ts]: Outdated — this targeted the earlier revision's `linkedElsewhere` heuristic, which no longer exists. Since 922e692 every bare user-attachments URL line is replaced with an inline `<video>` player, so duplicates each render a player and nothing is dropped. Covered by the new repeated-bare-URL regression test in 7bfb42b.

[the remark this answers is not in the document]
Comment (chhoumann) [reply src/gui/UpdateModal/UpdateModal.ts]: Outdated — the `linkedElsewhere` includes() check was removed in the 922e692 rewrite; bare URL lines now always become inline players (duplicates included), and thumbnails are only folded into the poster when they explicitly link the same URL. Regression test added in 7bfb42b.

File: src/gui/UpdateModal/UpdateModal.ts [hunk 2]
```diff
@@ -124,7 +156,7 @@ export class UpdateModal extends Modal {
 
 		const contentDiv = contentEl.createDiv("quickadd-update-modal");
 		const releaseNotes = this.releases
-			.map((release) => release.body)
+			.map((release) => tidyVideoAttachmentUrls(release.body))
```
[the remark this answers is not in the document]
Comment (chhoumann) [reply src/gui/UpdateModal/UpdateModal.ts]: Good catch — still applied to the rewrite. Fixed in 7bfb42b: `Release.body` is now typed `string | null` and the call site renders `release.body ?? ""`, restoring the old graceful degradation.
